controllers/vspheredeploymentzone_controller_domain.go

```diff
@@ -84,6 +84,16 @@ func (r vsphereDeploymentZoneReconciler) reconcileTopology(ctx *context.VSphereD
 		}
 	}
 
+	for _, networkConfig := range topology.NetworkConfigurations {
+		if networkConfig.NetworkName == "" {
```

Comment:
Should we validate for `NetworkName` being set instead of silently ignoring the array entry?

Reply:
My question actually is: should we make NetworkName required? Per https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/pull/1967/files#diff-2f616e5f5f40526609f3723e82cbcf01b16b9f240ea00fcdf3fc06b6aa6d164cR548 we just merge if it is not empty. This would allow a case where a failure domain has different nameservers, but same Portgroup name.

OTOH, there's no point IMO in validate a failure domain network config if the portgroup is not configured, so probably marking Network Name as required would be better IMHO.

WDYT?

Reply:
If the other code which overwrites stuff from `NetworkConfig` to the `vmNetworkDeviceSpec` and is ok with not replacing the NetworkName, we should drop the if here and keep it non-mandatory.

If that is not a use case, (and I assume it isn't according your above comment), then we should mark the NetworkName as required so we ensure on api side already that it cannot be misconfigured (by not setting the NetworkName).

Happy to also hear thoughts from @yastij or others :-)

Reply:
I think is better to make the networkName as a `key` inside the network array. Because if there is Multiple devices
 configured vsphereMachineTemplate, if the use forgot to specify the network name, it might accidentally configure Devices config on the vsphereMachineTemplate.